redis/src/script.rs

```diff
@@ -212,12 +212,44 @@ impl<'a> ScriptInvocation<'a> {
         cmd
     }
 
+    fn estimate_buflen(&self) -> usize {
+        self
+            .keys
+            .iter()
+            .chain(self.args.iter())
+            .fold(0, |acc, e| acc + e.len())
+            + 7 /* "EVALSHA".len() */
+            + self.script.hash.len()
+            + 1 /* Require a little more since rustc seems to reallocate vecs when length matches exactly. */
```

Comment:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=26ff6d3d5ddfaf6ad99aeb0c11c20679

I don't see this happening. Can you please explain?

Reply:
weird...

Reply:
I found the real reason is the length of keys.